src/file/file.repository.ts

```diff
@@ -106,19 +196,27 @@ export default class FileRepository {
    * @param populate - an option to populate the retrieved file's fields.
    * @param select - select certain fields of the files.
    */
-  static find(condition?: Object, populate?: string | Object, select?: string): Promise<IFile[]> {
+  static async find(condition?: Object, populate?: string | Object, select?: string): Promise<IFile[]> {
 
-    let findPromise = fileModel.find(condition);
+    let findPromise = baseFileModel.find(condition);
     if (populate) {
       findPromise = findPromise.populate(populate);
     }
     if (select) {
       findPromise = findPromise.select(select);
     }
-
-    return findPromise.exec().then((result) => {
-      return (result ? result.map((mongoObject => mongoObject.toObject())) : result);
+    const fulfilledFiles: IFile[] = [];
+    const rejectedFiles: IFile[] = [];
+    const files = await baseFileModel.find(condition).exec();
+    const setteledFiles = await Promise.allSettled(files.map(this.baseFileToIFile));
+    setteledFiles.forEach((file) => {
+      if (file.status === 'fulfilled' && file.value) fulfilledFiles.push(file.value);
+      else rejectedFiles.push((file as PromiseRejectedResult).reason);
```

Comment:
This is much better than the examples above

Reply:
done